interfaces/builtin/dbus_test.go

```diff
@@ -256,6 +299,200 @@ slots:
 	c.Assert(interfaces.BeforePrepareSlot(s.iface, slot), IsNil)
 }
 
+func (s *DbusInterfaceSuite) TestSanitizeSlotActivatableOk(c *C) {
+	var mockSnapYaml = `name: dbus-snap
+version: 1.0
+slots:
+  dbus-service-slot:
+    interface: dbus
+    bus: session
+    name: org.dbus-snap.session
+apps:
+  app-dbus-slot:
+    daemon: simple
+    daemon-scope: user
+    activates-on:
+    - dbus-service-slot
+`
+	info := snaptest.MockInfo(c, mockSnapYaml, nil)
+	slot := info.Slots["dbus-service-slot"]
+	c.Assert(interfaces.BeforePrepareSlot(s.iface, slot), IsNil)
+}
+
+func (s *DbusInterfaceSuite) TestSanitizeSlotActivatableNotDaemon(c *C) {
+	var mockSnapYaml = `name: dbus-snap
+version: 1.0
+slots:
+  dbus-service-slot:
+    interface: dbus
+    bus: session
+    name: org.dbus-snap.session
+apps:
+  app-dbus-slot:
+    activates-on:
+    - dbus-service-slot
+`
+	info := snaptest.MockInfo(c, mockSnapYaml, nil)
+	slot := info.Slots["dbus-service-slot"]
+	err := interfaces.BeforePrepareSlot(s.iface, slot)
+	c.Check(err, ErrorMatches, "only daemons can be activatable D-Bus services")
```

Comment:
This highlights my suggestion to handle this via snap validation. It would come up early (at snapcraft time) and does not require any dynamic connection state to determine.